sys/kern/kern_sig.c

```diff
@@ -1859,6 +1859,19 @@ sigaltstack_cheri_revoke(struct thread *td,
 			    (void * __capability)cheri_revoke_cap(sp);
 		}
 	}
+
+	if ((td->td_pflags & TDP_SIGFASTBLOCK) != 0) {
+		uintcap_t sbp = (uintcap_t)td->td_sigblock_ptr;
+
+		if (cheri_gettag(sbp)) {
```

Comment:
Would it make sense to have a helper function for use in places like this to revoke a single cap?  We duplicate this same block of code in various places for all the kernel hoarders.  Maybe I will work on that as a followup.

Reply:
Yeah, I nearly did that, but figured that was for another PR...

Reply:
Yeah, fine to not do it in this series.